test/functional/wiki_controller_test.rb

```diff
@@ -504,6 +504,18 @@ def teardown
     assert_response :success
   end
 
+  test "redirect to non-existent page fails gracefully; no redirect" do
+    wiki = node(:wiki_page)
+    slug = wiki.path.gsub('/wiki/', '')
+    user = rusers(:bob)
+    tagname = "redirect:nonsense"
+    wiki.add_tag(tagname, user)
+    assert_equal wiki.power_tag('redirect'), "nonsense"
```

Comment:
then we would change this to `assert !wiki.has_tag('redirect:nonsense')` to be sure the tag doesn't exist.